MSAL/src/native_auth/controllers/sign_in/MSALNativeAuthSignInController.swift

```diff
@@ -51,12 +52,13 @@ final class MSALNativeAuthSignInController: MSALNativeAuthTokenController, MSALN
             requestProvider: tokenRequestProvider,
             cacheAccessor: cacheAccessor,
             factory: factory,
-            responseValidator: tokenResponseValidator
+            responseValidator: tokenResponseValidator,
+            application: application
         )
     }
 
-    convenience init(config: MSALNativeAuthConfiguration, cacheAccessor: MSALNativeAuthCacheInterface) {
-        let factory = MSALNativeAuthResultFactory(config: config, cacheAccessor: cacheAccessor)
+    convenience init(config: MSALNativeAuthConfiguration, cacheAccessor: MSALNativeAuthCacheInterface, application: MSALNativeAuthPublicClientApplication?) {
+        let factory = MSALNativeAuthResultFactory(config: config, cacheAccessor: cacheAccessor, application: application)
```

Comment:
Personally, I'm not a fan of using a reference of the publicClientApplication, but I don't have a better solution for this. So, no complaint.
The only thing that I suggest is to have this instance only in the classes that we need it. If the application instance is only needed to create this factory, I'd add this factory as parameter in the convenience init.
I'm also wondering why we don't use the ControllerFactory to create an instance of the SignInController in the SignUp and ResetPassword controller. If we do, we could avoid sharing around the codebase this "application" instance.

Reply:
Addressed. Please take a look on the new PR; https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/2159